srml/system/src/lib.rs

```diff
@@ -490,10 +501,15 @@ impl<T: Trait> Module<T> {
 
 		let number = <Number<T>>::take();
 		let parent_hash = <ParentHash<T>>::take();
-		let mut digest = <Digest<T>>::take();
+		let digest = <PreDigest<T>>::take();
+		assert_eq!(digest.logs().len(), <NumDigests<T>>::take(), "Pre-digests are not altered by the runtime");
```

Comment:
I don't see why it's useful at all, if one can alter `PreDigest` you can also alter `NumDigests` to make that check pass.